File: app/src/main/java/info/upump/jym/entity/ExerciseDescription.java
package info.upump.jym.entity;


import java.util.Objects;

public class ExerciseDescription {
    private long id;
    private String img;
    private String title;
    private String defaultImg;

    public String getDefaultImg() {
        return defaultImg;
    }

    public void setDefaultImg(String defaultImg) {
        this.defaultImg = defaultImg;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getImg() {
        return img;
    }

    public void setImg(String img) {
        this.img = img;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ExerciseDescription that = (ExerciseDescription) o;
        return Objects.equals(img, that.img) &&
                Objects.equals(title, that.title) &&
                Objects.equals(defaultImg, that.defaultImg);
    }

    //без ID чтобы правильно сравнивать обьекты из текущей базы и базы бекапа
    @Override
    public int hashCode() {
        return Objects.hash(img, title, defaultImg);
    }

    @Override
    public String toString() {
        return "ExerciseDescription{" +
                "id=" + id +
                ", img='" + img + '\'' +
                ", title='" + title + '\'' +
                '}';
    }
}

File: mycompose/src/main/java/info/upump/mycompose/models/entity/ExerciseDescription.kt
package info.upump.mycompose.models.entity

import info.upump.database.entities.ExerciseDescriptionEntity
import java.util.Objects

class ExerciseDescription(
    var id: Long = 0,
    var img: String = "",
    var title: String = "",
    var defaultImg: String = ""
) {
    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as ExerciseDescription
        return img == that.img && title == that.title && defaultImg == that.defaultImg
    }

    //без ID чтобы правильно сравнивать обьекты из текущей базы и базы бекапа
    override fun hashCode(): Int {
        return Objects.hash(img, title, defaultImg)
    }

    override fun toString(): String {
        return "ExerciseDescription{" +
                "id=" + id +
                ", img='" + img + '\'' +
                ", title='" + title + " $defaultImg}"
    }

    companion object {
        fun mapFromDbEntity(entity: ExerciseDescriptionEntity): ExerciseDescription {
            val exerciseDescription = ExerciseDescription(entity._id)
            exerciseDescription.img = entity.img ?: ""
            exerciseDescription.defaultImg = entity.default_img!!
            exerciseDescription.title = entity.title!!

            return exerciseDescription
        }
    }
}